### PR TITLE
ENG-13955-CLEAN : A cleaner branch for ENG-13955

### PR DESCRIPTION
**Pull request by Yusarin**

### CODE REVIEW

File: src/frontend/org/voltdb/InvocationDispatcher.java [hunk 1]
```diff
@@ -519,8 +519,16 @@ else if ("@UpdateLogging".equals(procName)) {
...
 
-    private final boolean isSchemaEmpty() {
-        return m_catalogContext.get().database.getTables().size() == 0;
+    private final boolean toLoadSchemaFromSnapshot() {
+        CatalogMap<Table> tables = m_catalogContext.get().database.getTables();
+        if(tables.size() == 0) return true;
+        boolean ret = true;
+        for(Table t : tables) {
+            if(!t.getSignature().startsWith("VOLTDB_AUTOGEN_XDCR")) {
+                ret = false;
```
Comment (manjujames) [inline src/frontend/org/voltdb/InvocationDispatcher.java]: We can return false here immediately. We don't need to go through the rest of the tables.

File: src/frontend/org/voltdb/InvocationDispatcher.java [hunk 2]
```diff
@@ -519,8 +519,16 @@ else if ("@UpdateLogging".equals(procName)) {
         return null;
     }
 
-    private final boolean isSchemaEmpty() {
-        return m_catalogContext.get().database.getTables().size() == 0;
+    private final boolean toLoadSchemaFromSnapshot() {
```
Comment (manjujames) [inline src/frontend/org/voltdb/InvocationDispatcher.java]: May be rename this method to shouldLoadSchemaFromSnapshot.

File: tests/frontend/org/voltdb/regressionsuites/TestRestoreSysprocSuite.java [hunk 3]
```diff
@@ -199,6 +205,116 @@ public void testSaveAndPartialRestoreWithNonExistingExcludeTables() throws Excep
...
+                        "V1 INTEGER NOT NULL," +
+                        "V2 INTEGER NOT NULL," +
+                                          ");" +
+                       "CREATE TABLE object2(" +
+                        "ID INTEGER NOT NULL," +
+                        "C1 INTEGER NOT NULL," +
+                        "C2 INTEGER NOT NULL," +
+                                           ");";
+
+            cluster1Client.callProcedure("@AdHoc", simpleSchema);
```
Comment (manjujames) [inline tests/frontend/org/voltdb/regressionsuites/TestRestoreSysprocSuite.java]: Verify that this returns SUCCESS.

**Comment by manjujames** [comment]
Looks OK now other than one minor comment.

File: tests/frontend/org/voltdb/regressionsuites/TestRestoreSysprocSuite.java [hunk 4]
```diff
@@ -199,6 +205,123 @@ public void testSaveAndPartialRestoreWithNonExistingExcludeTables() throws Excep
...
+            cr = cluster1Client.callProcedure("@AdHoc", "insert into object1 values (1133,4,5);");
+            assertEquals(ClientResponse.SUCCESS, cr.getStatus());
+            cr = cluster1Client.callProcedure("@AdHoc", "insert into object2 values (1143,6,8);");
+            assertEquals(ClientResponse.SUCCESS, cr.getStatus());
+            TestSaveRestoreSysprocSuite.saveTablesWithDefaultOptions(cluster1Client, TESTNONCE);
+            TestSaveRestoreSysprocSuite.validateSnapshot(true, TESTNONCE);
+            m_config.shutDown();
+            m_config.startUp();
+
+//            String cluster2Root = "/tmp/" + System.getProperty("user.name") + "-cluster2";
```
Comment (luyangco) [inline tests/frontend/org/voltdb/regressionsuites/TestRestoreSysprocSuite.java]: Nitpick, please remove commented code from the pull request.

**Pull request merged**